### PR TITLE
Checkpoint update to wes2cromwell

### LINKED ISSUE #3877: Update nomenclature
See [#57](https://github.com/ga4gh/workflow-execution-service-schemas/pull/57)

### PR DESCRIPTION
**Pull request by geoffjentry**
As per standup today I'm opening up a PR for the work in progress and will transition to more targeted PRs after this

There's a lot in this PR which is just renaming and/or moving stuff around, but feel free to comment fully on everything since I disallowed it the last round. Be aware that my response might wind up being "that's already one of the things I want to do" and thus get pushed off to a followup PR.

Specific changes I made:

- Converted the actor based model for a future based model
- Updated the URL structure to reflect the [current nomenclature](https://github.com/ga4gh/workflow-execution-service-schemas/pull/33), closes #3877 
- Where possible I named things after the corresponding concept in the WES spec, which is most of the renaming here
- A few other small changes which were either new-to-WES or not-yet-implemented

### CODE REVIEW

File: wes2cromwell/src/main/scala/wes2cromwell/WesRunRoutes.scala [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+  implicit def materializer: ActorMaterializer
+
+  val log: LoggingAdapter
+
+  def cromwellUrl: URL
+  def cromwellApiVersion: String
+  def cromwellPath: URL = new URL(cromwellUrl.toString + s"/api/workflows/$cromwellApiVersion")
+
+  implicit val duration: FiniteDuration = ConfigFactory.load().as[FiniteDuration]("akka.http.server.request-timeout")
+  implicit lazy val timeout: Timeout = duration
```
Comment (Horneth) [inline wes2cromwell/src/main/scala/wes2cromwell/WesRunRoutes.scala]: Not sure why one is lazy and not the other ?

File: wes2cromwell/src/main/scala/wes2cromwell/RunListResponse.scala [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package wes2cromwell
+
+import spray.json.{JsonFormat, JsonParser}
+
+case class RunListResponse(runs: List[WesRunStatus], next_page_token: String)
+
+object RunListResponse {
+  import RunListResponseJsonSupport._
+
+  def toCromwellQueryResponse(json: String): RunListResponse = {
```
Comment (Horneth) [inline wes2cromwell/src/main/scala/wes2cromwell/RunListResponse.scala]: should this be renamed to `fromJson` like in `CromwellMetadata` ?

**Comment by geoffjentry** [comment]
@kshakir [way ahead of you](https://github.com/broadinstitute/cromwell/issues/2423)

**Comment by geoffjentry** [comment]
Spotted a problem on this, gonna fix & commit before merging

**Pull request merged**